src/Raven.Server/Smuggler/Documents/Iteration/IterationState.cs

```diff
@@ -0,0 +1,37 @@
+﻿using Raven.Server.ServerWide.Context;
+using Raven.Server.Utils.Enumerators;
+using Sparrow;
+using Voron.Impl;
+
+namespace Raven.Server.Smuggler.Documents.Iteration
+{
+    public abstract class IterationState<T> : PulsedEnumerationState<T>
+    {
+        protected IterationState(DocumentsOperationContext context)
+        {
+            Context = context;
+        }
+
+        protected readonly Size PulseLimit = new Size(16, SizeUnit.Megabytes); // TODO arek - make it configurable
+
+        protected readonly DocumentsOperationContext Context;
+
+        public int ReadCount { get; set; }
+
+        public override bool ShouldPulseTransaction()
+        {
+            if (ReadCount == 3)
```

Comment:
`== 3` ?  I assume that this is debug code right now?

Reply:
Yes, committed accidentally.